src/config/validate.rs

```diff
@@ -0,0 +1,103 @@
+use std::path::PathBuf;
+
+use super::{CliArgs, FileOrStdio, FileConfig};
+
+pub trait ValidatePath {
+    /// Validates and canonicalizes this PathBuf to point to a valid folder
+    fn validate_dir(&mut self, error_str: &str);
+    /// Validate and canonicalize this input, checking that the file exists
+    fn validate_file_exists(&mut self, error_str: &str);
+    /// Validate and canonicalize this input, ignoring if the file exists or not
+    fn validate_file_ignore(&mut self, error_str: &str);
+}
```

Comment:
These methods are panicking, not returning Results. Renaming them to `expect_*` would be clearer.